docs/how-to/create-mtls-client-credentials.md

```diff
@@ -1,170 +1,175 @@
+juju run self-signed-certificates/0 get-ca-certifictae
 ```
 
-### Create keystore (client cert)
+The result would be like below:
+
+```text
+Running operation 3 with 1 task
+  - task 4 on unit-self-signed-certificates-0
+
+Waiting for task 4...
+ca-certificate: |-
+  -----BEGIN CERTIFICATE-----
+  MIIDTTCCAjWgAwIBAgIUVKaEiZ0fKnWDOvlap16dZtA6fWkwDQYJKoZIhvcNAQEL
+  BQAwLDEqMCgGA1UEAwwhc2VsZi1zaWduZWQtY2VydGlmaWNhdGVzLW9wZXJhdG9y
+  MB4XDTI1MDYxOTA4NTgyOFoXDTI2MDYxOTA4NTgyOFowLDEqMCgGA1UEAwwhc2Vs
+  Zi1zaWduZWQtY2VydGlmaWNhdGVzLW9wZXJhdG9yMIIBIjANBgkqhkiG9w0BAQEF
+  AAOCAQ8AMIIBCgKCAQEAnKCNGgoe/d60W/QGKnWuzFQ+S/1ol2+GaMzg7eyklq2h
+  8zcT5YxkNIdY91awre09gvOERp3rrhumXgj72igufKjAoGn+M+xYiTC3cyiQq5SR
+  2f7UvgCvufawNBeQEFMOJ/Oih3aBJujOUx4wTO2AX6G7+U7JLfqJEjW1ZzoMGJx1
+  5Keyn4oXWTfylUkF+1QS6Nb5NNpjx5iFeSmCNU6i/P5p37m4xbb9L5r2feepU3N/
+  iESqlLhEDUtvV0/IXVCIe23Dx5tNxqZoe4DlQTK1NxJ5Zb25c2lV6MfRS57T2nGU
+  0VIMEkbrJ7Sc2CJnFhqJYR+81xGFtGjdfkC4d/AnRwIDAQABo2cwZTAfBgNVHQ4E
+  GAQWBBRV19sdiADtRwc4UFUej0ao03KrGzAhBgNVHSMEGjAYgBYEFFXX2x2IAO1H
+  BzhQVR6PRqjTcqsbMA4GA1UdDwEB/wQEAwICpDAPBgNVHRMBAf8EBTADAQH/MA0G
+  CSqGSIb3DQEBCwUAA4IBAQBSe2nUHoLA5Snn7R+r/Jp+agBjFAHT0LslULG0z7/s
+  GCo2/W84q2KOlDP1kUJ0C6JBeS1BZTd/ZzAuHBmCWkwOOmQnHCvzT3vxdLuKOab1
+  tdGZg0nvxiU1FDSkTchMccmeUMRk3aKzrYUMNg2PLxl2u0GdJhmYjhtARUte3nzo
+  ufPtyMx80lOEK02O8vzgwYidVr7xplbg8SdLKbGwMLH03Wv3w7ew9kN743HbT8AM
+  nx4xSLCybz3upJpRMXXkIenf7Rr7eDp3s4deAbGcvCo6B/XDeBOkSv8Sl7CsSejm
+  05qe06cC/6/K45CzOrWRwr4q5m6ENK/UT5fuOLFIoVJS
+  -----END CERTIFICATE-----
+```
 
-Create a client cert signed by the server
+Copy the certificate content into a file named `server.pem` and save it. You can also do that using a single command:
 
 ```bash
-# ---------- Keystore
-# create new private key --> client_key.pem
-openssl genrsa -out client_key.pem 4096
+juju run self-signed-certificates/0 get-ca-certificate --format json | jq -r '."self-signed-certificates/0"."results"."ca-certificate"' > server.pem
+```
+
+## Create mutual trust relation: client -> server
+
+Depending on the type of the client application, there might be different ways to trust. In this guide, we are using the console-based apps shipped with the `charmed-kafka` snap, which depend on java keystore/trustsores. Follow the steps below to create necessary Java keystore and truststore artifacts for the client application:
 
-# create new csr --> client_csr.pem
-openssl req -new -key client_key.pem -out client_csr.pem -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=$KAFKA_CLIENT_MTLS_CN"
+### Create client's keystore
 
-# sign new csr using new client CA --> client_cert.pem
-openssl x509 -req -CA ss_ca.pem -CAkey ss_ca.key -in client_csr.pem -out client_cert.pem -days 365 -CAcreateserial -passin pass:$SS_KEY_PASSWORD
+Create the client's keystore using the following commands:
 
-# create new chain --> client_chain.pem
-cat ss_ca.pem client_cert.pem client_key.pem > client_chain.pem
+```bash
+# Java keystore and truststore passwords
```

Comment:
Suggestion: Since we are editing this document, can we please make sure not to use comments in code blocks unless it's absolutely necessary? I feel like all comments in code blocks from this page can be placed before each respectable code block as instructions (potentially splitting a code block into multiple smaller ones). That is especially relevant for `bash`/`shell` type code blocks.